diff --git a/node_modules/react-native-agora/android/src/main/java/com/syan/agora/AgoraManager.java b/node_modules/react-native-agora/android/src/main/java/com/syan/agora/AgoraManager.java
index 2cb177d..18e19c3 100755
--- a/node_modules/react-native-agora/android/src/main/java/com/syan/agora/AgoraManager.java
+++ b/node_modules/react-native-agora/android/src/main/java/com/syan/agora/AgoraManager.java
@@ -1,7 +1,7 @@
 package com.syan.agora;
 
 import android.content.Context;
-import android.support.v7.widget.LinearLayoutCompat;
+import androidx.appcompat.widget.LinearLayoutCompat;
 import android.util.Log;
 import android.util.SparseArray;
 import android.view.SurfaceView;
diff --git a/node_modules/react-native-agora/android/src/main/java/com/syan/agora/AgoraModule.java b/node_modules/react-native-agora/android/src/main/java/com/syan/agora/AgoraModule.java
index de69b2f..50285f2 100644
--- a/node_modules/react-native-agora/android/src/main/java/com/syan/agora/AgoraModule.java
+++ b/node_modules/react-native-agora/android/src/main/java/com/syan/agora/AgoraModule.java
@@ -1,7 +1,7 @@
 package com.syan.agora;
 
 import android.graphics.Rect;
-import android.support.annotation.Nullable;
+import androidx.annotation.Nullable;
 import android.view.SurfaceView;
 
 import com.facebook.react.bridge.Arguments;
diff --git a/node_modules/react-native-agora/android/src/main/java/com/syan/agora/AgoraVideoView.java b/node_modules/react-native-agora/android/src/main/java/com/syan/agora/AgoraVideoView.java
index 8bc226f..73f0a43 100644
--- a/node_modules/react-native-agora/android/src/main/java/com/syan/agora/AgoraVideoView.java
+++ b/node_modules/react-native-agora/android/src/main/java/com/syan/agora/AgoraVideoView.java
@@ -7,7 +7,7 @@ import android.media.MediaCodecList;
 import android.media.MediaFormat;
 import android.media.MediaMuxer;
 import android.os.Build;
-import android.support.annotation.RequiresApi;
+import androidx.annotation.RequiresApi;
 import android.util.AttributeSet;
 import android.util.Log;
 import android.view.View;
diff --git a/node_modules/react-native-agora/android/src/main/java/com/syan/agora/AgoraViewManager.java b/node_modules/react-native-agora/android/src/main/java/com/syan/agora/AgoraViewManager.java
index 89ddb5a..0d2ed1a 100644
--- a/node_modules/react-native-agora/android/src/main/java/com/syan/agora/AgoraViewManager.java
+++ b/node_modules/react-native-agora/android/src/main/java/com/syan/agora/AgoraViewManager.java
@@ -44,7 +44,11 @@ public class AgoraViewManager extends SimpleViewManager<AgoraVideoView> {
 
     @ReactProp(name = "zOrderMediaOverlay")
     public void setZOrderMediaOverlay(final AgoraVideoView agoraVideoView, boolean zOrderMediaOverlay) {
-        surfaceView.setZOrderMediaOverlay(zOrderMediaOverlay);
+        if (surfaceView != null) {
+            surfaceView.setZOrderMediaOverlay(zOrderMediaOverlay);
+        } else{
+            agoraVideoView.setZOrderMediaOverlay(zOrderMediaOverlay);
+        }
     }
 
     @ReactProp(name = "remoteUid")
