diff --git a/node_modules/react-native-device-battery/react-native-device-battery.podspec b/node_modules/react-native-device-battery/react-native-device-battery.podspec
index 2becc23..9a0a8f9 100644
--- a/node_modules/react-native-device-battery/react-native-device-battery.podspec
+++ b/node_modules/react-native-device-battery/react-native-device-battery.podspec
@@ -10,7 +10,7 @@ Pod::Spec.new do |s|
 
   s.authors      = package['author']
   s.homepage     = package['homepage']
-  s.platforms    = { :ios => "9.0", :osx => "10.14" }
+  s.platforms    = { :ios => "12.4", :osx => "10.14" }
 
   s.source       = { :git => "https://github.com/robinpowered/react-native-device-battery.git", :tag => "v#{s.version}" }
   s.source_files  = "ios/**/*.{h,m}"
