diff --git a/node_modules/react-native-agora/android/src/main/java/com/syan/agora/AgoraViewManager.java b/node_modules/react-native-agora/android/src/main/java/com/syan/agora/AgoraViewManager.java
index 89ddb5a..4c343d2 100644
--- a/node_modules/react-native-agora/android/src/main/java/com/syan/agora/AgoraViewManager.java
+++ b/node_modules/react-native-agora/android/src/main/java/com/syan/agora/AgoraViewManager.java
@@ -44,7 +44,11 @@ public class AgoraViewManager extends SimpleViewManager<AgoraVideoView> {
 
     @ReactProp(name = "zOrderMediaOverlay")
     public void setZOrderMediaOverlay(final AgoraVideoView agoraVideoView, boolean zOrderMediaOverlay) {
-        surfaceView.setZOrderMediaOverlay(zOrderMediaOverlay);
+	if (surfaceView != null) {
+            surfaceView.setZOrderMediaOverlay(zOrderMediaOverlay);
+        } else{
+            agoraVideoView.setZOrderMediaOverlay(zOrderMediaOverlay);
+        }
     }
 
     @ReactProp(name = "remoteUid")
